### ISSUE TITLE
Save Multiple Lora as Metadata

### ISSUE THREAD

**Issue by DeepSeaCatz** [alexopus/ComfyUI-Image-Saver#44]
Hi there,

I'm having an issue where I'd wish to save multiple Loras that I'm stacking. I am not sure if the Image Saver node was meant to only save the checkpoint or if it could accept other types of models to populate Civitai resources. Here is my current workflow:
![workflow (17)](https://github.com/user-attachments/assets/02386136-0041-421f-b39a-d623dd852df7)

Currently all other metadata seems to be checked on Civitai :
![image](https://github.com/user-attachments/assets/7c69be2b-b97d-4c5f-b0e3-c52a888f0dd2)

I've tried to follow the json file format for json and adds commas between each Loras and add the main checkpoint at first. I am almost sure that this is where I would need to reword it so the modelname would accept it. I am not a fan of including all the `<lora:Lora:0.8>` inside the prompts, so that would be pretty useful:
![image](https://github.com/user-attachments/assets/12e4c478-ed55-439f-8e3c-f0a02f40810e)

I am currently testing this with XL, but will eventually tackle the Flux compatibility. Any help would be appreciated as I currently have to upload 1 image per resource used and this would be a game changer!

Thank you!

**Comment by alexopus** [comment]
Hashes for LoRAs and embeddings are extracted from the prompt, in the format that you can find in the readme. I recommend using ImpactWildcardEncode to do this conveniently. It also has advantage that you can easily copy and reuse LoRA info just by copying the string, instead of having to copy a bunch of widget settings. And you have to do this for embeddings already.

For the moment this is the only supported way and is unlikely to change. If you want to use a stacker from the efficiency pack, I believe there is a node that can convert that to a compatible string or you can do so manually.

**Comment by DeepSeaCatz** [comment]
Ah I see now, that make sense.

I've now fixed this with a **LoRA Stack to String converter** node and combined the main checkpoint with a **text concatenate**. I'd thought you had to feed them all in the modelname, but I've Text Concatenate it into the main prompt and it worked.
![image](https://github.com/user-attachments/assets/70a70404-d8bf-4fb7-be34-7e33b68f7a09)

While waiting, I as working on the 2nd node that I knew could also save metadata, the [RvTools](https://github.com/Rvage0815/ComfyUI-RvTools/issues/2). I was successful with their node and liked their "loras" input that I could feed in instead of in the prompt. Knowing that I could feed the Loras into the prompt earlier and I would have sticked to Image Saver nodes.

I am currently trying to make a workflow for Flux, that would also capture the metadata just like SDXL, but I am seeing lots of image quality difference when using a KSampler instead of the SamplerCustomAdvanced node to be able to gather the info. The thing with the SamplerCustomAdvanced node is that there are unique inputs that aren't compatible with both of the Image Save nodes and RvTools. I tried feeding the Flux Guidance number as a CFG, but I think the Guider is what I am missing as it seem to additionally mix with the models once more.

Were you successful in making having the Image Saver compatible with Flux before while having the same quality?
Here's my current workflow, which I've unfortunately used the RvTools nodes, but are very similar to the Image Saver nodes. On the left is the SamplerCustomAdvanced and the right is the KSampler results:
![378219845-2b57b059-b20b-4b07-ae3c-f594bbc72141](https://github.com/user-attachments/assets/7cdd0732-2bd7-4ba9-92ee-0bfab14f8c9d)
